File: Java-CodeServer/src/main/resources/js/prompto/error/IndexOutOfRangeError.js
var ExecutionError = require("./ExecutionError").ExecutionError;

function IndexOutOfRangeError() {
	ExecutionError.call(this);
	return this;
}

IndexOutOfRangeError.prototype = Object.create(ExecutionError.prototype);
IndexOutOfRangeError.prototype.constructor = IndexOutOfRangeError;

IndexOutOfRangeError.prototype.getExpression = function(context) {
	return context.getRegisteredValue("INDEX_OUT_OF_RANGE");
};

exports.IndexOutOfRangeError = IndexOutOfRangeError;

File: Java-Core/src/main/java/prompto/error/IndexOutOfRangeError.java
package prompto.error;

import prompto.expression.CategorySymbol;
import prompto.expression.IExpression;
import prompto.grammar.Identifier;
import prompto.runtime.Context;


public class IndexOutOfRangeError extends ExecutionError {

	private static final long serialVersionUID = 1L;

	@Override
	public IExpression getExpression(Context context) {
		return context.getRegisteredValue(CategorySymbol.class, new Identifier("INDEX_OUT_OF_RANGE"));
	}
}
